gobblin-api/src/main/java/gobblin/commit/CommitSequence.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.commit;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+import org.apache.commons.lang3.reflect.ConstructorUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Lists;
+
+import lombok.Getter;
+
+
+/**
+ * A sequence of {@link CommitStep}s that should be executed atomically.
+ *
+ * <p>
+ *    Typical pattern for building a {@link CommitSequence} with two {@link CommitStep}s:
+ *
+ *    <pre>
+ *      {@code
+ *        CommitSequence sequence = CommitSequence.newBuilder()
+ *                                  .withJobName(jobName)
+ *                                  .withDatasetUrn(datasetUrn)
+ *
+ *                                  .buildStep(FsRenameCommitStep.Builder.class)
+ *                                  .withProps(props)
+ *                                  .from(srcPath)
+ *                                  .to(dstPath)
+ *                                  .stepDone()
+ *
+ *                                  .buildStep(DatasetStateCommitStep.Builder.class)
+ *                                  .withProps(props)
+ *                                  .withDatasetUrn(datasetUrn)
+ *                                  .withDatasetState(datasetState)
+ *                                  .stepDone()
+ *
+ *                                  .build();
+ *      }
+ *    </pre>
+ * </p>
+ *
+ * @author ziliu
+ */
+public class CommitSequence {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CommitSequence.class);
```

Comment:
Seems you are using `@Slf4j` for some classes. Please make sure the use is consistent.